flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/DeltaJoinITCase.scala

```diff
@@ -106,34 +118,18 @@ class DeltaJoinITCase(enableCache: Boolean) extends StreamingTestBase {
       "+I[2.0, 2, 2022-02-02T02:02:02, 2, 2.0, 2022-02-02T02:02:02]",
       "+I[1.0, 1, 2021-01-01T01:01:01, 1, 1.0, 2021-01-01T01:01:01]"
     )
-    testUpsertResult(List("a1"), List("b1"), data1, data2, "a1 = b1 and a2 = b2", expected, 6)
-  }
-
-  @TestTemplate
-  def testJoinKeyNotContainsIndex(): Unit = {
-    val data1 = List(
-      changelogRow("+I", Double.box(1.0), Int.box(1), LocalDateTime.of(2021, 1, 1, 1, 1, 1)),
-      changelogRow("+I", Double.box(2.0), Int.box(2), LocalDateTime.of(2022, 2, 2, 2, 2, 2)),
-      // mismatch
-      changelogRow("+I", Double.box(3.0), Int.box(3), LocalDateTime.of(2023, 3, 3, 3, 3, 3))
-    )
 
-    val data2 = List(
-      changelogRow("+I", Int.box(1), Double.box(1.0), LocalDateTime.of(2021, 1, 1, 1, 1, 11)),
-      changelogRow("+I", Int.box(2), Double.box(2.0), LocalDateTime.of(2022, 2, 2, 2, 2, 22)),
-      // mismatch
-      changelogRow("+I", Int.box(99), Double.box(99.0), LocalDateTime.of(2099, 2, 2, 2, 2, 2))
-    )
-
-    // TestValuesRuntimeFunctions#KeyedUpsertingSinkFunction will change the RowKind from
-    // "+U" to "+I"
-    val expected = List("+I[1.0, 1, 2022-02-02T02:02:02, 1, 1.0, 2022-02-02T02:02:22]")
-
-    // could not optimize into delta join because join keys do not contain indexes strictly
-    assertThatThrownBy(
```

Comment:
I removed these expected failed tests because these exceptions have been tested in `DeltaJoinTest`. And I hope the IT tests can purely test the dataset, not for plan.